leptos_dom/src/node_ref.rs

```diff
@@ -2,38 +2,48 @@ use crate::{html::ElementDescriptor, HtmlElement};
 use leptos_reactive::{create_render_effect, signal_prelude::*};
 use std::cell::Cell;
 
-/// Contains a shared reference to a DOM node created while using the `view`
-/// macro to create your UI.
-///
-/// ```
-/// # use leptos::{*, logging::log};
-/// use leptos::html::Input;
-///
-/// #[component]
-/// pub fn MyComponent() -> impl IntoView {
-///     let input_ref = create_node_ref::<Input>();
-///
-///     let on_click = move |_| {
-///         let node =
-///             input_ref.get().expect("input_ref should be loaded by now");
-///         // `node` is strongly typed
-///         // it is dereferenced to an `HtmlInputElement` automatically
-///         log!("value is {:?}", node.value())
-///     };
-///
-///     view! {
-///       <div>
-///       // `node_ref` loads the input
-///       <input _ref=input_ref type="text"/>
-///       // the button consumes it
-///       <button on:click=on_click>"Click me"</button>
-///       </div>
-///     }
-/// }
-/// ```
-#[repr(transparent)]
-pub struct NodeRef<T: ElementDescriptor + 'static>(
-    RwSignal<Option<HtmlElement<T>>>,
+  cfg_if::cfg_if!(
+  if #[cfg(debug_assertions)] {
+    #[allow(missing_docs)]
+    pub struct NodeRef<T: ElementDescriptor + 'static> {
+      element: RwSignal<Option<HtmlElement<T>>>,
+      defined_at: &'static std::panic::Location<'static>,
+    }
+  } else {
+    /// Contains a shared reference to a DOM node created while using the `view`
+    /// macro to create your UI.
+    ///
+    /// ```
+    /// # use leptos::{*, logging::log};
+    /// use leptos::html::Input;
+    ///
+    /// #[component]
+    /// pub fn MyComponent() -> impl IntoView {
+    ///     let input_ref = create_node_ref::<Input>();
+    ///
+    ///     let on_click = move |_| {
+    ///         let node =
+    ///             input_ref.get().expect("input_ref should be loaded by now");
+    ///         // `node` is strongly typed
+    ///         // it is dereferenced to an `HtmlInputElement` automatically
+    ///         log!("value is {:?}", node.value())
+    ///     };
+    ///
+    ///     view! {
+    ///       <div>
+    ///       // `node_ref` loads the input
+    ///       <input _ref=input_ref type="text"/>
+    ///       // the button consumes it
+    ///       <button on:click=on_click>"Click me"</button>
+    ///       </div>
+    ///     }
+    /// }
+    /// ```
+    #[repr(transparent)]
```

Comment:
It's not clear to me that there's any benefit to `#[repr(transparent)]`. If you think it's important, I'd propose that rather than a `cfg_if` block that removes the docs in debug mode, this should be `#[cfg_attr(not(debug_assertions), repr(transparent)]` if possible, and `#[cfg(debug_assertions)]` on the `defined_at` field.

Reply:
> It's not clear to me that there's any benefit to #[repr(transparent)]

I just providing context - not expressing an opinion..
When I "git blame" this line I am directed to this issue - [831](https://github.com/leptos-rs/leptos/pull/831)

With this subheading 

> The goal of all these changes is to reduce WASM binary size with opt-level = "z" and reduce the number of allocations for tiny objects.

All very normal  -- I am going to try and keep it rather than throw it out in favor of better debug